controllers/singlePageCrawler.js

```diff
@@ -25,14 +25,24 @@ async function scrapeSingleURL(productURL) {
       const description = $('div.product--description').text();
 
       console.log(title, description, img, price, numPrice);
-      const crawledProduct = new Product({
+      const updatedProduct = {
         url: productURL,
         title: title,
         description: description,
         img: img,
         price: numPrice
-      });
-      await crawledProduct.save();
+      };
+      console.log(updatedProduct);
+      const doc = await Product.findOne({ _id: id });
```

Comment:
This line is not required.
You can directly use `id` in your `updateOne`.

```
await Product.updateOne(
        { _id: id },
        {
          title: updatedProduct.title,
          description: updatedProduct.description,
          img: updatedProduct.img,
          price: updatedProduct.price
        }
      );
```

Reply:
I was aware of this one. Thought it would be syntactic sugar to see what happens behind the curtains. But it's obviously smarter to solve this in less code.